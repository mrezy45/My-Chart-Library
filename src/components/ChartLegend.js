import React from 'react';
import '../styles/ChartStyles.css';

const ChartLegend = ({ data, colors = [] }) => {
  return (
    <div className="chart-legend">
      {data.map((d, i) => (
        <div key={i} className="legend-item">
          <span
            className="legend-color"
            style={{ backgroundColor: colors[i % colors.length] || 'steelblue' }}
          ></span>
          <span className="legend-label">{d.label}</span>
        </div>
      ))}
    </div>
  );
};

export default ChartLegend;