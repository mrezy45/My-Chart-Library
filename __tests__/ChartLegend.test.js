import React from 'react';
import { render } from '@testing-library/react';
import ChartLegend from '../src/components/ChartLegend';

describe('ChartLegend Component', () => {
  it('renders without crashing', () => {
    const data = [
      { label: 'January', value: 40 },
      { label: 'February', value: 60 },
    ];
    const { container } = render(<ChartLegend data={data} colors={['#ff0000', '#00ff00']} />);
    expect(container).toMatchSnapshot();
  });

  it('displays the correct number of legend items', () => {
    const data = [
      { label: 'January', value: 40 },
      { label: 'February', value: 60 },
      { label: 'March', value: 20 },
    ];
    const { getAllByText } = render(<ChartLegend data={data} colors={['#ff0000', '#00ff00', '#0000ff']} />);
    const items = getAllByText(/January|February|March/);
    expect(items.length).toBe(data.length);
  });
});