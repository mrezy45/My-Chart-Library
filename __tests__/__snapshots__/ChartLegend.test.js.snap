// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChartLegend Component renders without crashing 1`] = `
<div>
  <div
    class="chart-legend"
  >
    <div
      class="legend-item"
    >
      <span
        class="legend-color"
        style="background-color: rgb(255, 0, 0);"
      />
      <span
        class="legend-label"
      >
        January
      </span>
    </div>
    <div
      class="legend-item"
    >
      <span
        class="legend-color"
        style="background-color: rgb(0, 255, 0);"
      />
      <span
        class="legend-label"
      >
        February
      </span>
    </div>
  </div>
</div>
`;
